### ISSUE TITLE
Implementing VariableDeclaration

### ISSUE THREAD

**Issue by joshuavial** [uittorio/ts-auto-mock#84]
related to #83 - when I setup the following test

```javascript
import { createMock } from 'ts-auto-mock';
import { Connection, Model, Document} from 'mongoose';

describe('for mongoose', () => {
    describe('connection', () => {
        interface Thing extends Document{
          name: string
        }
        interface Interface {
          thing: Model<Thing>
        }

        fit('should set an empty string', () => {
            const mock: Interface = createMock<Interface>();
            console.log(mock.thing);
        });
    });
});

```

It gives the following error
```
Child process failed to process the request: TypeError: Cannot read property '0' of undefined
    at /ts-auto-mock/dist/transformer/index.js:1:6932
```
running it with jest-ts-auto-mock also helpfully reveals 
`NOT IMPLEMENTED VariableDeclaration`

I'm happy to have a crack at implementing VariableDeclaration if that would be useful.

**Comment by joshuavial** [comment]
```javascript
import { createMock } from 'ts-auto-mock';
import { Connection, Model, Document} from 'mongoose';

describe('for mongoose', () => {
    describe('connection', () => {
        interface Thing extends Document{
          name: string
        }
        interface Interface {
          thing: Model<Thing>
        }

        fit('should set an empty string', () => {
            const mock: Interface = createMock<Interface>();
            console.log(mock.thing);
        });
    });
});

```

**Comment by uittorio** [comment]
Hi, I will start working on this issue soon.
The generics implementations to fix the majority of missing implementations of mongoose is in code reivew. 
This issue is a bit tricky because it introduce multiple declaration for the same type
In this instance

```ts 
var Collection: Collection;
interface Collection {}
```
I need to first understand what is this scenario, if its intended by typescript (I believe so) and in that case select the correct declaration :)

**Comment by uittorio** [comment]
This issue has been resolved in version 1.3.1
